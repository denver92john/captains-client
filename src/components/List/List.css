.list-component {
    padding: 20px;
    background-color: ghostwhite;
    min-width: 200px;
    min-height: 200px;
}

.named-list {
    list-style-type: none;
    text-align: left;
}

.named-list-item {
    border-bottom: 1px solid black;
	padding: 5px;
	display: flex;
	justify-content: space-between;
}

.named-list-item:last-of-type {
    border-bottom: none;
}

.list-item-button {
    width: 20px;
}