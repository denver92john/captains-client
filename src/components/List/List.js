import React, {Component} from 'react';
import FormList from '../FormList/FormList';
import ListItem from '../ListItem/ListItem';
import './List.css';

/*
<ListItem key={item.id} item_name={item.item_name} link={null} />
<ListItem key={item.id} item_name={item.list_name} link={`/list/${item.id}/list-items`} />
*/

class List extends Component {
    render() {
        const {items, onPost} = this.props;
        return (
            <div className="wrapper list-component">
                <FormList onPost={onPost} />
                <ul className="named-list">
                    {items.map(item =>
                        <ListItem 
                            key={item.id}
                            //item_name={item.list_name}
                            item={item}
                            link={`/list/${item.id}/list-items`}
                        />
                    )}
                </ul>
            </div>
        );
    }
}

export default List;