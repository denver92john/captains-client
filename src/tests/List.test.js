import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import List from '../components/List/List';

describe('List component', () => {
    const itemsProp = [
        {
            id: 1,
            list_name: 'Joe',
            user_id: 1
        },
        {
            id: 2,
            list_name: 'Green',
            user_id: 1
        }
    ]

    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <List items={itemsProp} />
            </BrowserRouter>,
            div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI as expected', () => {
        const tree = renderer
            .create(<BrowserRouter>
                    <List items={itemsProp} />
                </BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})