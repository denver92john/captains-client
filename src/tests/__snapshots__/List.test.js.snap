// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders UI as expected 1`] = `
<div
  className="wrapper list-component"
>
  <form
    className="form-list"
  >
    <input
      className="input form-list-input"
      name="item_name"
      required={true}
      type="text"
    />
    <button
      className="form-list-button"
      type="submit"
    />
  </form>
  <ul
    className="named-list"
  >
    <li
      className="list-item"
    >
      <a
        className="list-item-link"
        href="/list/1/list-items"
        onClick={[Function]}
      >
        Joe
      </a>
      <div>
        <button
          className="formitem-button"
          onClick={[Function]}
          type="button"
        />
        <button
          className="formitem-button"
          onClick={[Function]}
          type="button"
        />
      </div>
    </li>
    <li
      className="list-item"
    >
      <a
        className="list-item-link"
        href="/list/2/list-items"
        onClick={[Function]}
      >
        Green
      </a>
      <div>
        <button
          className="formitem-button"
          onClick={[Function]}
          type="button"
        />
        <button
          className="formitem-button"
          onClick={[Function]}
          type="button"
        />
      </div>
    </li>
  </ul>
</div>
`;
